File: app/src/main/java/com/dialogapp/dialog/model/Post.java
package com.dialogapp.dialog.model;

/**
 * Custom Post model used by the views
 */

public class Post {

    public final long id;
    public final String url;
    public final String contentHtml;
    public final String datePublished;

    public final String post_property_dateRelative;
    public final Boolean post_property_isDeletable;
    public final Boolean post_property_isFavorite;

    public final String author_name;
    public final String author_url;
    public final String author_avatar_url;
    public final String author_info_username;

    public Post(long id, String url, String contentHtml, String datePublished,
                String post_property_dateRelative, Boolean post_property_isDeletable,
                Boolean post_property_isFavorite,
                String author_name, String author_url, String author_avatar_url, String author_info_username) {
        this.id = id;
        this.url = url;
        this.contentHtml = contentHtml;
        this.datePublished = datePublished;
        this.post_property_dateRelative = post_property_dateRelative;
        this.post_property_isDeletable = post_property_isDeletable;
        this.post_property_isFavorite = post_property_isFavorite;
        this.author_name = author_name;
        this.author_url = author_url;
        this.author_avatar_url = author_avatar_url;
        this.author_info_username = author_info_username;
    }
}

File: app/src/main/java/com/dialogapp/dialog/model/Post.kt
package com.dialogapp.dialog.model

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import com.squareup.moshi.Json

@Entity(tableName = "posts", primaryKeys = ["id", "belongsToEndpoint"], foreignKeys = [ForeignKey(
        entity = EndpointData::class,
        parentColumns = ["endpoint"],
        childColumns = ["belongsToEndpoint"])], indices = [Index(value = ["belongsToEndpoint"], unique = false)]
)
data class Post(@field:Json(name = "id")
                @Json(name = "id")
                val id: String,
                @field:Json(name = "content_html")
                @Json(name = "content_html")
                val contentHtml: String,
                @field:Json(name = "url")
                @Json(name = "url")
                val url: String,
                @field:Json(name = "date_published")
                @Json(name = "date_published")
                val datePublished: String,
                @field:Json(name = "author")
                @Json(name = "author")
                @Embedded(prefix = "author_")
                val author: Author,
                @field:Json(name = "_microblog")
                @Json(name = "_microblog")
                @Embedded(prefix = "post_property_")
                val microblog: PostProperties) {

    var belongsToEndpoint: String = ""

    data class PostProperties(@field:Json(name = "date_relative")
                              @Json(name = "date_relative")
                              val dateRelative: String,
                              @field:Json(name = "is_deletable")
                              @Json(name = "is_deletable")
                              val isDeletable: Boolean,
                              @field:Json(name = "is_favorite")
                              @Json(name = "is_favorite")
                              val isFavorite: Boolean,
                              @field:Json(name = "is_conversation")
                              @Json(name = "is_conversation")
                              val isConversation: Boolean)
}